crypto/keys/src/traits.rs

```diff
@@ -2,19 +2,41 @@
 
 //! Abstract traits used by Structs which implement key management
 
-use crate::{Digest, LengthMismatch, ReprBytes, Unsigned};
-use alloc::{string::String, vec::Vec};
+use crate::{Digest, LengthMismatch, ReprBytes};
+
 use core::{fmt::Debug, hash::Hash};
 use displaydoc::Display;
 use mc_crypto_digestible::Digestible;
 use mc_util_from_random::FromRandom;
 use rand_core::{CryptoRng, RngCore};
+
+#[cfg(feature = "alloc")]
+use alloc::{vec::Vec};
+
+#[cfg(feature = "serde")]
 use serde::{de::DeserializeOwned, Deserialize, Serialize};
 
+
+/// Marker trait for serialization when `serde` feature is enabled
+#[cfg(feature = "serde")]
+pub trait MaybeSerde: DeserializeOwned + Serialize {}
```

Comment:
this is very clever, kudos

Reply:
you may be able to tell i have experienced this problem before 🤣